core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala

```diff
@@ -583,3 +587,23 @@ private class FirehoseListenerThatAcceptsSparkConf(conf: SparkConf) extends Spar
     case _ =>
   }
 }
+
+private class SlowDeserializable(var readDuration: Int)
```

Comment:
Probably easier to make the sleep time hardcoded? Then a lot of this code goes away.

(And if doing that, the `@throws` annotations are most probably unnecessary.)